### DISCUSSION TITLE
404 instead 422

### DISCUSSION THREAD

**Discussion by psprokofiev** [spatie/laravel-data#391] [General]
Data class
```
class StoredMedia extends Data
{
    public function __construct(
        #[Validation\Required, Validation\StringType]
        public string $project,
    ) {
        //
    }
}
```

API Controller method
```
public function store(StoredMedia $data)
{
    dd($data->toArray());
}
```

Request with post method return `404 Not Found` instead `422` status.

If I remove string with Validation rules and replace `public string $project` with `public ?string $project`, all is ok.

* Laravel Framework 10.4.1
* spatie/laravel-data 3.2

**Comment by rubenvanassche** [comment]
Could you provide me a Pest test (take a look at `DataTest` for inspiration) which reproduces this problem?

**Comment by psprokofiev** [comment]
Sorry, I don't have experience with Pest yet, how can I help you another way?

**Reply by rubenvanassche** [reply]
What data are you exactly providing?

**Reply by psprokofiev** [reply]
For example, empty requesy, without any parameters.

**Reply by rubenvanassche** [reply]
I don't think this is a data problem since we don't send out 404's, could you provide me a laravel project showing this issue?

**Comment by psprokofiev** [comment]
Huh, all is ok, I added header "accept: application/json" to request...
